clang/test/CIR/CodeGen/binop.cpp

```diff
@@ -32,13 +32,10 @@ void b1(bool a, bool b) {
 
 // CHECK: cir.ternary(%3, true
 // CHECK-NEXT: %7 = cir.load %1
-// CHECK-NEXT: cir.ternary(%7, true
-// CHECK-NEXT: cir.const #true
-// CHECK-NEXT: cir.yield
-// CHECK-NEXT: false {
-// CHECK-NEXT: cir.const #false
-// CHECK-NEXT: cir.yield
-// CHECK: cir.yield
+// CHECK-NEXT: %8 = cir.const #true
+// CHECK-NEXT: %9 = cir.const #false
+// CHECK-NEXT: %10 = cir.select if %7 then %8 else %9
```

Comment:
Looks like some `cir.select` fold patterns are missing so that this code does not further simplify:

- `cir.select %x then #true else #false` -> `%x`
- `cir.select %x then #false else #true` -> `cir.unary not %a`
- `cir.select %x then %a else %a` -> `%a`

Will add later in a separate PR.